src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,93 @@
+using Flux
+
+function basicblock()
+  dilation = 1;
+  if dilation > 1
+    error("Dilation > 1 not supported in Basic Block");
+  end
+  basic =  Chain(
+      Conv((3, 3), inplanes=>planes, stride=stride, pad=(1,1), dilation=dilation),
+      BatchNorm(planes, λ = relu),
+      Conv((3, 3), planes=>planes, stride=stride, pad=(1,1), dilation=dilation),
+      BatchNorm(planes)
+    );
+  return basic
+end
+
+function bottleneck(planes::Int, stride::Int)
+  stride = 2
+  base_width = 64;
+  groups = 1;
+  width = Int(planes * (base_width / 64.)) * groups;
+  residual =  Chain(
+          Conv((1, 1), inplanes=>width, stride=(2,2)),
+          BatchNorm(width, λ=relu),
+          Conv((3, 3), width=>width, stride=(2,2), pad=(1,1), dilation=dilation),
+          BatchNorm(width, λ=relu),
+          Conv((1, 1), width=>planes * expansion, stride=(2,2)),
+          BatchNorm(planes * expansion, λ=relu)
+          );
+  return residual
+end 
+
+function identity(inplanes::Int, planes::Int, stride::Int, expansion::Int)
+  inplanes = inplanes;
+  planes = planes;
+  downsample =  Chain(Conv((1, 1), inplanes=>planes * expansion, stride=stride),
+  BatchNorm(planes * expansion, λ=relu));
+  return identity
+end
+
+function resnet()
+  planes = 64;
+  inplanes = 64;
+  expansion = 1;
+  stride = 1;
+  layers = Chain(
+      Conv((7, 7), 3=>planes, stride=(2, 2), pad=(3, 3)),
+      BatchNorm(planes, λ=relu),
+      MaxPool((3, 3), stride=(2, 2), pad=(1, 1))
+      );
+    
+  push!(layers, SkipConnection(layers, basicblock(inplanes, planes, stride));
+  stride, dilation = basicblock();
+
+  for i in 2:4
+    stride = 2;
+    push!(layers, SkipConnection(layers, basicblock(inplanes, planes, stride));
+    push!(layers, SkipConnection(layers, identity(inplanes, planes, stride, 1)));
+    planes = planes*2;
+    inplanes = planes * expansion;
+    expansion = 4;
```

Comment:
I don't understand this logic with `expansion`. But fix the other issues first, then we can come back to it.

Reply:
expansion is one of the metrics used in a calculation for testing if  downsampling will happen or not. for ``bottleneck`` it is 4 and for ``basicblock`` it is 1.